frontend/src/metabase/actions/containers/ActionCreator/InlineActionSettings.tsx

```diff
@@ -0,0 +1,146 @@
+
+type PublicWritebackAction = WritebackAction & {
+  public_uuid: string;
+};
+
+interface OwnProps {
+  onClose: () => void;
+  actionId: number;
+}
+
+interface EntityLoaderProps {
+  action: WritebackAction;
+}
+
+interface StateProps {
+  siteUrl: string;
+  createPublicLink: ({ id }: { id: number }) => void;
+  deletePublicLink: ({ id }: { id: number }) => void;
+}
+
+type ActionSettingsInlineProps = OwnProps & EntityLoaderProps & StateProps;
+
+export const ActionSettingsTriggerButton = ({
+  onClick,
+}: {
+  onClick: () => void;
+}) => (
+  <Tooltip tooltip={t`Action settings`}>
+    <Button onlyIcon onClick={onClick} icon="gear" iconSize={16} />
+  </Tooltip>
+);
+
+const InlineActionSettings = ({
+  onClose,
+  actionId,
+  action,
+  siteUrl,
+  createPublicLink,
+  deletePublicLink,
+}: ActionSettingsInlineProps) => {
+  const id = useUniqueId();
+  const isPublic = isActionPublic(action);
+
+  const [isModalOpen, { turnOn: openModal, turnOff: closeModal }] = useToggle();
+
+  const handleTogglePublic = (isPublic: boolean) => {
+    if (isPublic) {
+      createPublicLink({ id: actionId });
+    } else {
+      openModal();
+    }
+  };
+
+  const handleDisablePublicLink = () => {
+    deletePublicLink({ id: actionId });
+  };
+
+  return (
+    <ActionSettingsContainer>
+      <SidebarContent title={t`Action settings`} onClose={onClose}>
+        <ActionSettingsContent>
+          <ToggleContainer>
+            <span>
+              <ToggleLabel htmlFor={id}>{t`Make public`}</ToggleLabel>
+              <Tooltip
+                tooltip={t`Creates a publicly shareable link to this action.`}
+              >
+                <Icon name="info" size={10} />
+              </Tooltip>
+            </span>
+            <Toggle id={id} value={isPublic} onChange={handleTogglePublic} />
+            <Modal isOpen={isModalOpen}>
+              <ConfirmContent
+                title={t`Disable this public link?`}
+                content={t`This will cause the existing link to stop working. You can re-enable it, but when you do it will be a different link.`}
+                onAction={handleDisablePublicLink}
+                onClose={closeModal}
+              />
+            </Modal>
+          </ToggleContainer>
+          {isPublic && (
+            <CopyWidgetContainer>
+              <CopyWidget
+                value={Urls.publicAction(siteUrl, action.public_uuid)}
+                aria-label={t`Public action link URL`}
+              />
+            </CopyWidgetContainer>
+          )}
+        </ActionSettingsContent>
+      </SidebarContent>
+    </ActionSettingsContainer>
+  );
+};
+
+const mapStateToProps = (state: State) => ({
+  siteUrl: getSetting(state, "site-url"),
+});
+
+const mapDispatchToProps = {
+  createPublicLink: Actions.actions.createPublicLink,
+  deletePublicLink: Actions.actions.deletePublicLink,
+};
+
+function isActionPublic(
+  action: WritebackAction,
+): action is PublicWritebackAction {
```

Comment:
Need to use TypeScript's type predicate so it knows that if the check passes then `public_uuid` will not be null. I think there might be a better way than creating another type here like I imagine using something like `Required<WritebackAction, "public_uuid">` or something.

Reply:
It's good enough IMO